pkg/client/rekor_client.go

```diff
@@ -64,3 +72,65 @@ func GetRekorClient(rekorServerURL string, opts ...Option) (*client.Rekor, error
 	registry.Add("signedCheckpoint", &util.SignedNote{}, util.SignedCheckpointValidator)
 	return client.New(rt, registry), nil
 }
+
+func GenerateTransparencyLogEntry(anon models.LogEntryAnon) (*rekor_pb.TransparencyLogEntry, error) {
+	logIDHash, err := hex.DecodeString(*anon.LogID)
+	if err != nil {
+		return nil, fmt.Errorf("decoding logID string: %w", err)
+	}
+
+	rootHash, err := hex.DecodeString(*anon.Verification.InclusionProof.RootHash)
+	if err != nil {
+		return nil, fmt.Errorf("decoding inclusion proof root hash: %w", err)
+	}
+
+	inclusionProofHashes := make([][]byte, len(anon.Verification.InclusionProof.Hashes))
+	for i, hash := range anon.Verification.InclusionProof.Hashes {
+		hashBytes, err := hex.DecodeString(hash)
+		if err != nil {
+			return nil, fmt.Errorf("decoding inclusion proof hash: %w", err)
+		}
+		inclusionProofHashes[i] = hashBytes
+	}
+
+	b, err := base64.StdEncoding.DecodeString(anon.Body.(string))
```

Comment:
Should we check that Body is a `string` and err out if not?

Reply:
Body is a base64 encoded string so it should always be a string